### PR TITLE
Document the `opam-repository` release process

### PR DESCRIPTION
**Pull request by CraigFe**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+- Check that no `.opam` files contain `pin-depends` fields. If so, release those
+  packages first.
+
+```sh
+; git grep -A 10 "pin-depends" *.opam
+```
+
+- Make a pull-request to this repository containing pre-release changes (drop
+  `pin-depends`, add release number to `CHANGES.md`, anu new constraints) and an
```
Comment (Ngoguey42) [inline RELEASE.md:34]:
```suggestion
  `pin-depends`, add release number to `CHANGES.md`, any new constraints) and an
```

**Review by Ngoguey42 — APPROVED** [review]

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+[dune-release]: https://github.com/ocamllabs/dune-release
+[opam-repo]: https://github.com/ocaml/opam-repository
+
+## Releasing to opam-repository and GitHub
+
+- Check that no `.opam` files contain `pin-depends` fields. If so, release those
+  packages first.
+
+```sh
+; git grep -A 10 "pin-depends" *.opam
```
Comment (samoht) [inline RELEASE.md:30]: Can you open an issue on dune-release to track this with `dune-release lint` ? /cc @pitag-ha

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+```sh
+; opam pin add odoc.2.0.0 https://github.com/ocaml/odoc.git  # Use a modern Odoc
+
+; dune-release tag                        #  Create appropriate Git tag by reading CHANGES.md
+; dune-release distrib --skip-tests       #  Build release archive
+; dune-release publish distrib --verbose  #  Push release archive to GitHub
+; dune-release publish doc --verbose      #  Push documentation to GitHub pages
+; dune-release opam pkg                   #  Generate `opam` files for `opam-repository`
+; dune-release opam submit                #  Make PR to `opam-repository`
```
Comment (samoht) [inline RELEASE.md:56]: what's wrong with:
```
$ dune-release tag  
$ dune-release
```
?

Comment (avsm) [reply RELEASE.md:56]: I do `dune-release tag; dune-release` as well

Comment (CraigFe) [reply RELEASE.md:56]: I use those for smaller repositories too. For Irmin, I find the `--skip-tests` and `--verbose` flags helpful, so tend towards the individual stages here.

Comment (samoht) [reply RELEASE.md:56]: Maybe this means we need a `dune-release --verbose --skip-tests` option?

File: RELEASE.md (hunk 2 above)
Comment (pitag-ha) [reply RELEASE.md:30]: That's a good point, thanks! I've just opened [that issue](https://github.com/ocamllabs/dune-release/issues/356) in dune-release.

So from what I understand the `dune-release` lints could catch if one has forgotten to remove the `pin-depends` field before releasing. This step @CraigFe is doing here of finding out which packages are in `pin-depends` (if any) and releasing them first, would still have to be performed manually though. Is that right?

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+[git-tags]: https://git-scm.com/book/en/v2/Git-Basics-Tagging
+[pages-docs]: https://mirage.github.io/irmin
+[dune-release]: https://github.com/ocamllabs/dune-release
+[opam-repo]: https://github.com/ocaml/opam-repository
+
+## Releasing to opam-repository and GitHub
+
+- Check that no `.opam` files contain `pin-depends` fields. If so, release those
+  packages first.
```
Comment (pitag-ha) [inline RELEASE.md:27]: What's your workflow of releasing those packages first? Do you run `dune-release -p <pkg_name(s)>`? If so, might it be worth mentioning that?

File: RELEASE.md (hunk 3 above)
Comment (pitag-ha) [reply RELEASE.md:56]: That's also a good point. I think `dune-release` allows the `--verbose` flag, but not the `skip-tests` one. I've opened [an issue](https://github.com/ocamllabs/dune-release/issues/357) also for that.

File: RELEASE.md (hunk 4 above)
Comment (CraigFe) [reply RELEASE.md:27]: The pin-depended packages are usually https://github.com/mirage/index/ and https://github.com/mirage/repr/, so I suppose the process is "recursively execute this process on those repositories".

**Comment by CraigFe** [comment]
Thanks @pitag-ha for submitting those issues :pray: (https://github.com/ocamllabs/dune-release/issues/356, https://github.com/ocamllabs/dune-release/issues/357)

File: RELEASE.md (hunk 2 above)
Comment (CraigFe) [reply RELEASE.md:30]: Yep, it'd still be manual. I suppose the corresponding feature here is a multi-repository release process, but that sounds like a big feature.

**Pull request merged**